src/main/kotlin/com/terraformation/backend/documentproducer/api/ValuesController.kt

```diff
@@ -92,17 +126,22 @@ class ValuesController(
       @PathVariable projectId: ProjectId,
       @RequestBody payload: UpdateVariableValuesRequestPayload
   ): SimpleSuccessResponsePayload {
-    val existingValueIds = payload.operations.mapNotNull { it.getExistingValueId() }
-    val existingBases = variableValueStore.fetchBaseProperties(existingValueIds)
-    val operations =
-        payload.operations.map { operationPayload ->
-          val base = operationPayload.getExistingValueId()?.let { existingBases[it] }
-          operationPayload.toOperationModel(projectId, base)
-        }
+    try {
+      val existingValueIds = payload.operations.mapNotNull { it.getExistingValueId() }
+      val existingBases = variableValueStore.fetchBaseProperties(existingValueIds)
+      val operations =
+          payload.operations.map { operationPayload ->
+            val base = operationPayload.getExistingValueId()?.let { existingBases[it] }
+            operationPayload.toOperationModel(projectId, base)
+          }
 
-    variableValueService.validate(operations)
+      variableValueService.validate(operations)
 
-    variableValueStore.updateValues(operations)
+      variableValueStore.updateValues(operations)
+    } catch (e: Exception) {
```

Comment:
Is there an error that you're expecting but unaware of?

Reply:
Oof, nope, good catch. That was left over from me playing with error handling as I developed the code.